packages/ui-components/src/components/toggle-tip/toggle-tip.tsx

```diff
@@ -0,0 +1,105 @@
+import { ETooltipPosition } from '../tooltip/tooltip.types';
+import { IToggleTip, IToggleTipEvents } from './toggle-tip.types';
+
+/**
+ * @part toggle-tip-container - The toggle tip container.
+ * @part toggle-tip-slot-content - The toggle tip slot content.
+ */
+@Component({
+	tag: 'kv-toggle-tip',
+	styleUrl: 'toggle-tip.scss',
+	shadow: true
+})
+export class KvToggleTip implements IToggleTip, IToggleTipEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) text!: string;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) icon: EIconName = EIconName.Info;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) position?: ETooltipPosition;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) allowedPositions?: ETooltipPosition[];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) options?: Partial<ComputePositionConfig>;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) isOpen?: boolean = false;
+
+	/** @inheritdoc */
+	@Event() openStateChange: EventEmitter<boolean>;
+
+	/** The Host's element reference */
+	@Element() el: HTMLKvToggleTipElement;
+
+	@Listen('click', { target: 'window' })
+	checkForClickOutside(ev: { target: Node }) {
+		if (this.el.contains(ev.target)) {
+			return;
+		}
+		const toggletip = this.el.shadowRoot.querySelector('#toggle-tip') as HTMLElement;
+		toggletip.style.display = '';
+		this.isOpen = !this.isOpen;
+		this.openStateChange.emit(this.isOpen);
+		this.calculateTogglePositions();
+	}
+
+	componentDidRender() {
+		this.calculateTogglePositions();
+	}
+
+	private calculateTogglePositions = () => {
+		const toggleIcon = this.el.shadowRoot.querySelector('#toggle-tip-content');
+		const toggletip = this.el.shadowRoot.querySelector('#toggle-tip') as HTMLElement;
+
+		const position = isEmpty(this.allowedPositions) ? this.position : undefined;
+		const middleware = [offset(DEFAULT_OFFSET), shift(DEFAULT_SHIFT_CONFIG)];
+
+		if (isEmpty(position)) {
+			middleware.push(
+				autoPlacement({
+					...DEFAULT_AUTOPLACEMENT_CONFIG,
+					allowedPlacements: this.allowedPositions
+				})
+			);
+		}
+
+		const onButtonClick = () => {
+			this.isOpen = !this.isOpen;
+			if (this.isOpen) {
+				toggletip.style.display = 'inline-block';
+				computePosition(toggleIcon, toggletip, {
+					...DEFAULT_TOOLTIP_CONFIG,
+					placement: position,
+					middleware
+				}).then(({ x, y }) => {
+					Object.assign(toggletip.style, {
+						left: `${x}px`,
+						top: `${y}px`
+					});
+				});
+			} else {
+				toggletip.style.display = '';
+			}
+			this.openStateChange.emit(this.isOpen);
+		};
+		toggleIcon.addEventListener('mousedown', onButtonClick, { once: true });
+	};
+
+	render() {
+		return (
+			<Host>
+				<kv-icon id="toggle-tip-content" name={this.icon} exportparts="icon" />
+				<div id="toggle-tip" class="toggle-tip-container" part="toggle-tip-container">
+					<div class="toggle-tip-text">{this.text}</div>
```

Comment:
this should be displayed only if `text` is defined

Reply:
<div class="toggle-tip-text">{this.text}</div> this here correct? i will remove the ! from text and place here a verification